homeassistant/components/unifiprotect/sensor.py

```diff
@@ -0,0 +1,408 @@
+        precision=2,
+    ),
+    ProtectSensorEntityDescription(
+        key=_KEY_RES_HD,
+        name="Resolution: HD Video",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:cctv",
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        ufp_value="storage_stats.storage_distribution.hd_usage.percentage",
+        precision=2,
+    ),
+    ProtectSensorEntityDescription(
+        key=_KEY_RES_4K,
+        name="Resolution: 4K Video",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:cctv",
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        ufp_value="storage_stats.storage_distribution.uhd_usage.percentage",
+        precision=2,
+    ),
+    ProtectSensorEntityDescription(
+        key=_KEY_RES_FREE,
+        name="Resolution: Free Space",
+        native_unit_of_measurement=PERCENTAGE,
+        icon="mdi:cctv",
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        ufp_value="storage_stats.storage_distribution.free.percentage",
+        precision=2,
+    ),
+    ProtectSensorEntityDescription(
+        key=_KEY_CAPACITY,
+        name="Recording Capacity",
+        native_unit_of_measurement=TIME_SECONDS,
+        icon="mdi:record-rec",
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        ufp_value="storage_stats.capacity",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors for UniFi Protect integration."""
+    data: ProtectData = hass.data[DOMAIN][entry.entry_id]
+    entities: list[ProtectDeviceEntity] = async_all_device_entities(
+        data,
+        ProtectDeviceSensor,
+        all_descs=ALL_DEVICES_SENSORS,
+        camera_descs=CAMERA_SENSORS,
+        sense_descs=SENSE_SENSORS,
+    )
+    entities += _async_nvr_entities(data)
+
+    async_add_entities(entities)
+
+
+@callback
+def _async_nvr_entities(
+    data: ProtectData,
+) -> list[ProtectDeviceEntity]:
+    entities: list[ProtectDeviceEntity] = []
+    device = data.api.bootstrap.nvr
+    for description in NVR_SENSORS:
+        entities.append(ProtectNVRSensor(data, device, description))
+        _LOGGER.debug("Adding NVR sensor entity %s", description.name)
+
+    return entities
+
+
+class SensorValueMixin(Entity):
+    """A mixin to provide sensor values."""
+
+    @callback
+    def _clean_sensor_value(self, value: Any) -> Any:
+        if isinstance(value, timedelta):
+            value = int(value.total_seconds())
+        elif isinstance(value, datetime):
+            # UniFi Protect value can vary slightly over time
+            # truncate to ensure no extra state_change events fire
+            value = value.replace(second=0, microsecond=0)
```

Comment:
truncate the seconds as well? That means the precision becomes a minute... that sounds a bit much imho?

Reply:
Yeah... unfortunately even the seconds vary over time. @bdraco did a bunch of debugging on this and truncating to the minute was the more reliable way since UFP overtime always gave different second/sub-second values. 

Otherwise, you could have like a state_changed almost every time the device updates.

Reply:
I found my old screenshot from when I was testing it before: 
![msedge_Jqfpg9sZvU](https://user-images.githubusercontent.com/490848/148308822-7ae7a0d1-42c5-41b7-a58c-6ac63c582843.png)

Reply:
alright 👍